src/Compute/Compute/VirtualMachine/Config/NewAzureVMConfigCommand.cs

```diff
@@ -381,25 +381,34 @@ public override void ExecuteCmdlet()
                 vm.PlatformFaultDomain = this.PlatformFaultDomain;
             }
 	    
-	    if (this.IsParameterBound(c => c.SecurityType) && this.SecurityType?.ToLower() != ConstantValues.StandardSecurityType)
+	        if (this.IsParameterBound(c => c.SecurityType) && this.SecurityType?.ToLower() != ConstantValues.StandardSecurityType)
             {
                 if (vm.SecurityProfile == null)
                 {
                     vm.SecurityProfile = new SecurityProfile();
                 }
-                if (vm.SecurityProfile.UefiSettings == null)
-                {
-                    vm.SecurityProfile.UefiSettings = new UefiSettings();
-                }
                 vm.SecurityProfile.SecurityType = this.SecurityType;
 
                 if (vm.SecurityProfile.SecurityType?.ToLower() == ConstantValues.TrustedLaunchSecurityType || vm.SecurityProfile.SecurityType?.ToLower() == ConstantValues.ConfidentialVMSecurityType)
                 {
+                    if (vm.SecurityProfile.UefiSettings == null)
+                    {
+                        vm.SecurityProfile.UefiSettings = new UefiSettings();
+                    }
                     vm.SecurityProfile.UefiSettings.VTpmEnabled = vm.SecurityProfile.UefiSettings.VTpmEnabled == null ? true : this.EnableVtpm;
                     vm.SecurityProfile.UefiSettings.SecureBootEnabled = vm.SecurityProfile.UefiSettings.SecureBootEnabled == null ? true : this.EnableSecureBoot;
                 }
             }
 
+            if (this.SecurityType != null && this.SecurityType?.ToLower() == ConstantValues.StandardSecurityType)
```

Comment:
what if this.SecurityType != null but not  this.SecurityType?.ToLower() == ConstantValues.StandardSecurityType

Reply:
Then the above code block would be executed. I think this could just be an Else If of the above.